Framework/PythonInterface/plugins/algorithms/RefineSatellitePeaks.py

```diff
@@ -0,0 +1,181 @@
+from mantid.kernel import *
+from mantid.api import *
+from mantid.simpleapi import *
+
+import numpy as np
+from scipy.spatial import KDTree
+from scipy.cluster.vq import kmeans2
+import scipy.cluster.hierarchy as hcluster
+
+
+class RefineSatellitePeaks(DataProcessorAlgorithm):
+
+    def category(self):
+        return 'Crystal\\Peaks'
+
+    def PyInit(self):
+        self.declareProperty(IPeaksWorkspaceProperty(name="MainPeaks",
```

Comment:
Is Main peaks a good name? wouldn't NuclearPeaks be better?, maybe something to ask the Instrument Scientists.